src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/MultipartFormDataRequestContentProvider.cs

```diff
@@ -0,0 +1,495 @@
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using AutoRest.CSharp.Common.Input;
+using AutoRest.CSharp.Common.Output.Expressions.KnownValueExpressions;
+using AutoRest.CSharp.Common.Output.Expressions.Statements;
+using AutoRest.CSharp.Common.Output.Expressions.ValueExpressions;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Generation.Writers;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Serialization;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Models.Types;
+using static AutoRest.CSharp.Common.Output.Models.Snippets;
+using MultipartFormDataContent = System.Net.Http.MultipartFormDataContent;
+
+namespace AutoRest.CSharp.Common.Output.Models.Types.HelperTypeProviders
+{
+    internal class MultipartFormDataRequestContentProvider: ExpressionTypeProvider
+    {
+        private static readonly Lazy<MultipartFormDataRequestContentProvider> _instance = new(() => new MultipartFormDataRequestContentProvider());
+        private class FormDataItemTemplate<T> { }
+        public static MultipartFormDataRequestContentProvider Instance => _instance.Value;
+        private MultipartFormDataRequestContentProvider() : base(Configuration.HelperNamespace, null)
+        {
+            DeclarationModifiers = TypeSignatureModifiers.Internal;
+            Inherits = Configuration.ApiTypes.RequestContentType;
+            _multipartContentField = new FieldDeclaration(
+                modifiers: FieldModifiers.Private | FieldModifiers.ReadOnly,
+                type: typeof(MultipartFormDataContent),
+                name: "_multipartContent");
+            _randomField = new FieldDeclaration(
+                modifiers: FieldModifiers.Private | FieldModifiers.Static | FieldModifiers.ReadOnly,
+                type: typeof(Random),
+                name: "_random",
+                initializationValue: New.Instance(typeof(Random), Array.Empty<ValueExpression>()),
+                serializationFormat: SerializationFormat.Default);
+            _boundaryValuesFields = new FieldDeclaration(
+                modifiers: FieldModifiers.Private | FieldModifiers.Static | FieldModifiers.ReadOnly,
+                type: typeof(char[]),
+                name: "_boundaryValues",
+                initializationValue: Literal("0123456789=ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz").ToCharArray(),
+                serializationFormat: SerializationFormat.Default);
+            ValueExpression contentTypeHeaderExpression = new UnaryOperatorExpression("!", new MultipartFormDataContentExpression(_multipartContentField).Property("Headers").Property("ContentType"), true);
+            ValueExpression contentTypeHeaderValueExpression = new InvokeInstanceMethodExpression(contentTypeHeaderExpression, nameof(MediaTypeHeaderValue.ToString), Array.Empty<ValueExpression>(), null, false);
```

Comment:
if this is the only place `contentTypeHeaderExpression` is used, we should inline it, since it is not so long right now.
And we should use this:
```suggestion
            ValueExpression contentTypeHeaderValueExpression = contentTypeHeaderExpression.InvokeToString();
```